### PR TITLE
Data Insights Refactor

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes:

Fixes <issue-number>

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/rest/DataInsightAPI.ts [hunk 1]
```diff
@@ -17,6 +17,23 @@ import { DataInsightChartResult } from '../generated/dataInsight/dataInsightChar
...
+  TotalDataAssetsSummaryCard = 'total_data_assets_summary_card',
+  DataAssetsWithDescriptionSummaryCard = 'data_assets_with_description_summary_card',
+  DataAssetsWithOwnerSummaryCard = 'data_assets_with_owner_summary_card',
+  TotalDataAssetsWithTierSummaryCard = 'total_data_assets_with_tier_summary_card',
+}
+
+export interface DataInsightCustomChartResult {
+  results: Array<{ count: number; day: number; group: string }>;
+}
+
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/rest/DataInsightAPI.ts:36]: aren't we are getting types from json schema?

File: openmetadata-ui/src/main/resources/ui/src/styles/components/button.less [hunk 2]
```diff
@@ -69,3 +69,9 @@ button {
 .ant-btn-primary {
   box-shadow: none;
 }
+
+.ant-btn.btn-outlined {
+  border: 1px solid @primary-color;
+  color: @primary-color;
+  background: transparent;
+}
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/styles/components/button.less:77]: .btn-outlined is custom class? if not this will impact across app, is that expected?

**Review by ShaileshParmar11 — COMMENTED** [review]
left few comment

File: openmetadata-ui/src/main/resources/ui/src/components/DataInsight/KPIChart.tsx [hunk 3]
```diff
@@ -210,50 +224,71 @@ const KPIChart: FC<Props> = ({
...
+                          valueFormatter={(value, key) => {
+                            const isPercentage = key
+                              ? mapKPIMetricType[key] ===
+                                KpiTargetType.Percentage
+                              : KpiTargetType.Number;
+
+                            return isPercentage
+                              ? round(Number(value), 2) + '%'
+                              : value + '';
+                          }}
```
[the remark this answers is not in the document]
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/DataInsight/KPIChart.tsx:252]: Will keep it in this file since it's depend on some computed value from KPI data

File: openmetadata-ui/src/main/resources/ui/src/styles/components/button.less (hunk 2 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/styles/components/button.less:77]: Yes it's custom for us

File: bootstrap/sql/migrations/native/1.5.0/mysql/postDataMigrationSQLScript.sql [hunk 4]
```diff
@@ -24,6 +24,80 @@ WHERE name IN (
...
+          'configSource', JSON_OBJECT(
+            'connection', JSON_EXTRACT(dbse.json, '$.connection.config.metastoreConnection'),
+            'appName', JSON_UNQUOTE(JSON_EXTRACT(dbse.json, '$.connection.config.appName'))
+          )
+        )
+      )
+    )
+  )
+  , '$.connection.config.appName'), '$.connection.config.metastoreConnection')
+WHERE dbse.serviceType = 'DeltaLake';
```
Comment (IceS2) [inline bootstrap/sql/migrations/native/1.5.0/mysql/postDataMigrationSQLScript.sql:45]: @ulixius9 I think something happened when updating the branch since this shouldn't be here. It's about another PR and it was already moved: https://github.com/open-metadata/OpenMetadata/pull/16789/files

File: bootstrap/sql/migrations/native/1.5.0/postgres/postDataMigrationSQLScript.sql [hunk 5]
```diff
@@ -23,11 +23,77 @@ WHERE name IN (
...
+SET json = JSONB_SET(
+  JSONB_SET(
+    json,
+    '{connection,config,configSource}',
+    JSONB_BUILD_OBJECT('connection', json->'connection'->'config'->'metastoreConnection')
+  ),
+  '{connection,config,configSource,appName}',
+  json->'connection'->'config'->'appName'
+) #- '{connection,config,metastoreConnection}' #- '{connection,config,appName}'
+WHERE serviceType = 'DeltaLake';
```
Comment (IceS2) [inline bootstrap/sql/migrations/native/1.5.0/postgres/postDataMigrationSQLScript.sql:37]: Same thing as above

**Review by mohityadav766 — APPROVED** [review]

File: bootstrap/sql/migrations/native/1.5.0/mysql/postDataMigrationSQLScript.sql (hunk 4 above)
Comment (ulixius9) [reply bootstrap/sql/migrations/native/1.5.0/mysql/postDataMigrationSQLScript.sql:45]: Oh I see might have messed the merge

**Review by ShaileshParmar11 — DISMISSED** [review]

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/registry.py [hunk 6]
```diff
@@ -44,6 +41,5 @@
 build_registry.add(PipelineType.dbt.value)(build_dbt_dag)
 build_registry.add(PipelineType.profiler.value)(build_profiler_dag)
 build_registry.add(PipelineType.TestSuite.value)(build_test_suite_dag)
-build_registry.add(PipelineType.dataInsight.value)(build_data_insight_dag)
```
Comment (pmbrull) [inline openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/registry.py:47]: On a separate PR, but remember to pop a docs change and highlight we are removing the python side of the DI workflow

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/search/opensearch/OpenSearchDataInsightsClient.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    Request request = new Request(method, path);
+    request.setJsonEntity(payload);
+
+    return client.performRequest(request);
+  }
+
+  @Override
+  public void createLifecyclePolicy(String name, String policy) throws IOException {
+    try {
+      performRequest("PUT", String.format("_plugins/_ism/policies/%s", name), policy);
```
Comment (pmbrull) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/search/opensearch/OpenSearchDataInsightsClient.java:32]: nit - should we have an Enum of possible supported requests?

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/CostAnalysisWorkflow.java [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+
+  @Getter private final WorkflowStats workflowStats = new WorkflowStats();
+
+  public CostAnalysisWorkflow(
+      Long timestamp, int batchSize, Optional<DataInsightsApp.Backfill> backfill) {
+    this.endTimestamp =
+        TimestampUtils.getEndOfDayTimestamp(TimestampUtils.subtractDays(timestamp, 1));
+    this.startTimestamp = TimestampUtils.getStartOfDayTimestamp(endTimestamp);
+
+    // TODO: Implement Backfill by using DataAsset Version.
```
Comment (pmbrull) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/CostAnalysisWorkflow.java:75]: are we tracking these TODOs in a ticket?

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/processors/AggregatedCostAnalysisReportDataProcessor.java [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+                    .withFourteenDays(
+                        (double) frequentlyUsedDataAssets.getCount().getFourteenDays() + 1));
+            frequentlyUsedDataAssets.withSize(
+                frequentlyUsedDataAssets
+                    .getSize()
+                    .withFourteenDays(
+                        (double) frequentlyUsedDataAssets.getSize().getFourteenDays() + tableSize));
+          }
+          // Compute 30Days
+          periodTimestamp =
```
Comment (pmbrull) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/processors/AggregatedCostAnalysisReportDataProcessor.java:203]: these Compute X days look super repetitive, not sure if we can have a loop/method to handle them?

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/processors/DatabaseServiceTablesProcessor.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+          Optional<AccessDetails> oAccessed =
+              Optional.ofNullable(oTableLifeCycleData.get().getAccessed());
+
+          if (oAccessed.isPresent()) {
+            oTableLifeCycle = oTableLifeCycleData;
+          }
+        }
+
+        // Get Size data.
+        // TODO: Does the DataInsightsProcess have access to PII?
```
Comment (pmbrull) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/processors/DatabaseServiceTablesProcessor.java:59]: we're not blocking PII to admins/bots for now

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityDAO.java [hunk 11]
```diff
@@ -323,7 +323,7 @@ List<String> migrationListAfterWithOffset(
       @Define("nameHashColumn") String nameHashColumnName,
       @Bind("limit") int limit);
 
-  @SqlQuery("SELECT json FROM <table> <cond> AND ORDER BY name LIMIT :limit OFFSET :offset")
+  @SqlQuery("SELECT json FROM <table> <cond> ORDER BY name LIMIT :limit OFFSET :offset")
```
Comment (pmbrull) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityDAO.java:326]: why is this needed? I guess it would impact everywhere right?

**Review by pmbrull — APPROVED** [review]
approving it since it's huge, left some nits/questions we can tackle in following PRs

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/processors/AggregatedCostAnalysisReportDataProcessor.java (hunk 9 above)
Comment (IceS2) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/processors/AggregatedCostAnalysisReportDataProcessor.java:203]: I thought about it but we did not touch the CostAnalysis or WebAnalytics schemas and flows.

Maybe there's a hacky way of matching the number of days to the specific attribute getters and setters but I didn't think it was worth the hussle since we should refactor the way we deal with Cost Analysis to make it a bit more flexible as well afaik.

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityDAO.java (hunk 11 above)
Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityDAO.java:326]: you are so right

Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityDAO.java:326]: not sure how this did not blow up anywhere else

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/CostAnalysisWorkflow.java (hunk 8 above)
Comment (IceS2) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/CostAnalysisWorkflow.java:75]: They are not, I'll track it in the Data Insights Design document at least!

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/search/opensearch/OpenSearchDataInsightsClient.java (hunk 7 above)
Comment (IceS2) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/search/opensearch/OpenSearchDataInsightsClient.java:32]: What's you ridea? Not sure I followed it

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/registry.py (hunk 6 above)
Comment (IceS2) [reply openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/registry.py:47]: Absolutely right. I'll add it to the TODO to be done before the release is done

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/search/opensearch/OpenSearchDataInsightsClient.java (hunk 7 above)
Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/search/opensearch/OpenSearchDataInsightsClient.java:32]: We have a lot of "PUT" strings in a bunch of places. We could have `HTTPMethod.PUT` instead, and same for DELETE< etc.

**Review by pmbrull — APPROVED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by pmbrull — DISMISSED** [review]

**Review by harshach — APPROVED** [review]
